src/test/java/org/opensearch/knn/index/NestedSearchIT.java

```diff
@@ -74,11 +74,45 @@ public void testNestedSearch_whenKIsTwo_thenReturnTwoResults() {
         Float[] queryVector = { 1f, 1f };
         Response response = queryNestedField(INDEX_NAME, 2, queryVector);
 
-        List<Object> hits = (List<Object>) ((Map<String, Object>) createParser(
+        Map<String, Object> hits = ((Map<String, Object>) createParser(
             MediaTypeRegistry.getDefaultMediaType().xContent(),
             EntityUtils.toString(response.getEntity())
-        ).map().get("hits")).get("hits");
-        assertEquals(2, hits.size());
+        ).map().get("hits"));
+
+        List<Object> internalHits = (List<Object>) hits.get("hits");
+        assertEquals(2, internalHits.size());
+
+        int totalHit = (Integer) ((Map<String, Object>) hits.get("total")).get("value");
+        assertEquals(2, totalHit);
+    }
+
+    @SneakyThrows
+    public void testNestedSearchWithFaiss_whenKIsTwo_thenReturnTwoResults() {
+        createKnnIndex(2, KNNEngine.FAISS.getName());
+
+        String doc1 = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+            .addVectors(FIELD_NAME_VECTOR, new Float[] { 1f, 1f }, new Float[] { 1f, 1f })
+            .build();
+        addKnnDoc(INDEX_NAME, "1", doc1);
+
+        String doc2 = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+            .addVectors(FIELD_NAME_VECTOR, new Float[] { 2f, 2f }, new Float[] { 2f, 2f })
+            .build();
+        addKnnDoc(INDEX_NAME, "2", doc2);
+
+        Float[] queryVector = { 1f, 1f };
+        Response response = queryNestedField(INDEX_NAME, 2, queryVector);
+
+        Map<String, Object> hits = ((Map<String, Object>) createParser(
+            MediaTypeRegistry.getDefaultMediaType().xContent(),
+            EntityUtils.toString(response.getEntity())
+        ).map().get("hits"));
+
+        List<Object> internalHits = (List<Object>) hits.get("hits");
+        assertEquals(2, internalHits.size());
+
+        int totalHit = (Integer) ((Map<String, Object>) hits.get("total")).get("value");
```

Comment:
There is a function in the KNNRestTestCase called as `parseTotalSearchHits` please use that.

Reply:
ack